src/audio/src/src.c

```diff
@@ -462,7 +462,7 @@ static struct comp_dev *src_new(const struct comp_driver *drv,
 	comp_cl_info(&comp_src, "src_new()");
 
 	/* validate init data - either SRC sink or source rate must be set */
-	if (ipc_src->source_rate == 0 && ipc_src->sink_rate == 0) {
+	if (ipc_src->source_rate == 0 || ipc_src->sink_rate == 0) {
 		comp_cl_err(&comp_src, "src_new(): SRC sink and source rate are not set");
 		return NULL;
 	}
```

Comment:
please change condition in error message also, `and` -> `or`

Reply:
HI @ktrzcinx : This is not the final solution. Once it's decided, will update the error messages accordingly.

Reply:
ok, can we have an inline comment (as a reminder) that this is intermediate solution and the final solution is pending on X.

Reply:
Yes. I will do that.

Reply:
Be careful! In topologies load in new() the the fixed side rate is set only, the variable side (from host or PCM params) is left to zero. Have you tested this in a real device with real topology?

Reply:
As far as I know there's no CI test for SRC. The testbench run overrides the topology for rates so it's not the same as IPC from linux kernel.

Reply:
@singalsu : Sorry for the late reply. I didn't test this on a real device with real topology. But, the logic i have used here (&& -> ||) is wrong. I have to change this.